containers/podman-desktop
# feat: allow to use libkrun and applehv machines starting from 5.2.0

### What does this PR do?

This PR allows to manage both applehv and libkrun machines at the same time starting from 5.2.0 .
Even though a provider is defined in the `containers.conf` you should be able to work with both applehv and libkrun machines (so create, start, stop, remove, edit ... actions).

The only problem is that libkrun only supports max 8 cpus. If you create a machine with 8+ cpus, you are able to create it but it will always fail to start. So, before creating a machine, I reduce the cpu value to 8 if it's larger. It would be good to edit the maximum value at the UI level but I didn't find an easy way so far. Any idea is appreciated.

If Podman is less than 5.1.0 or you are not on a Mac it should work as before.

In draft as I still have to try it on Windows and write tests but it's ready if you want to start playing with it

### Screenshot / video of UI

https://github.com/user-attachments/assets/fd13ca3c-a075-40a2-86fc-e03b325b966f

### What issues does this PR fix or reference?

it resolves #8084 

### How to test this PR?

1. edit the value here with the Podman version you are using https://github.com/containers/podman-desktop/compare/main...lstocchi:i8084?expand=1#diff-c885162dd627b7cd812c83544b19b9b166179fec79ddbc13273fede7c2566791R1792
2. you should see and being able to use both applehv and libkrun machines. When using the cli you will only have visibility on the current provider set.
